src/main/java/com/example/petbuddybackend/controller/websocket/ChatWebSocketController.java

```diff
@@ -96,45 +89,28 @@ public void handleSubscribeToMessageTopic(SessionSubscribeEvent event) {
     @EventListener
     public void handleUnsubscribeToMessageTopic(SessionUnsubscribeEvent event) {
         StompHeaderAccessor accessor = StompHeaderAccessor.wrap(event.getMessage());
-        String destination = accessor.getDestination();
-
-        if(!sessionContext.containsSubscriptionId(accessor.getSubscriptionId()) || sessionContext.isEmpty()) {
-            return;
-        }
-
-        String username = sessionContext.getUsername();
-        Long chatId = sessionContext.getChatId();
-        String sessionId = sessionContext.getSessionId();
-        String subscriptionId = accessor.getSubscriptionId();
-
-        chatSessionService.sendNotifications(chatId, new ChatNotificationLeft(chatId, username));
-        chatSessionService.unsubscribe(chatId, username, sessionId, subscriptionId);
+        String username = HeaderUtils.getUser(accessor);
+        chatSessionService.onUserUnsubscribe(username, accessor.getSubscriptionId());
 
         log.debug(
                 "Event unsubscribe at {}; sessionId: {}; user: {}",
                 URL_CHAT_TOPIC_BASE,
                 accessor.getSessionId(),
-                accessor.getUser() == null ? "null" : accessor.getUser().getName()
+                username
         );
     }
 
     @EventListener
     public void handleDisconnectFromWebSocket(SessionDisconnectEvent event) {
```

Comment:
rozumiem, ze obslugujemy ten event drugi raz, zeby zrobic dzialania dedykowane dla chatu?

Reply:
W sensie tam mam ten `onUserUnsubscribe` i potrzebuje to wywołać. Nie wiem czy o to chodzi w pytaniu

Reply:
Bo ten event i tak obsługujemy w innym kontrolerze, więc tutaj też obsługujemy żeby wydzielić logikę dotyczaca chatu?